src/Core/Framework/Adapter/Translation/Translator.php

```diff
@@ -61,6 +62,11 @@ class Translator extends AbstractTranslator
      */
     private array $snippets = [];
 
+    /**
+     * @var array<string, array<string, bool>>
+     */
+    private array $cacheTags = [];
```

Comment:
Maybe
```suggestion
    private array $alreadyDispatchedCacheTags = [];
```

a bit longer, but IMHO makes it easier to understand what it used for
same for the system config service

Reply:
BTW of course this is now outdated